source/slang/hlsl.meta.slang

```diff
@@ -6573,6 +6573,108 @@ matrix<T, N, M> sqrt(matrix<T, N, M> x)
     MATRIX_MAP_UNARY(T, N, M, sqrt, x);
 }
 
+// The following cbrt(float) and rcbrt(float) algorithms are based upon:
+//
+// Moroz L, Samotyy V, Walczyk CJ, Cieśliński JL.
+// Fast Calculation of Cube and Inverse Cube Roots Using a Magic Constant and Its Implementation on Microcontrollers.
+// Energies. 2021; 14(4):1058.
+// https://doi.org/10.3390/en14041058
+//
+// Published under the CC BY 4.0 license.
+//
+// They use no lookup tables, which makes them vectorizable in theory.
+
+__target_intrinsic(cuda, "$P_cbrt($0)")
+[__readNone]
+float cbrt(float x)
```

Comment:
We also need to provide a derivative of this in diff.meta.slang to make it differentiable, like other math functions.

Reply:
That will work, but the compiler generated derivatives won't be as efficient as hand coded ones.
My suggestion is to implement a custom derivative with `[ForwardDerivativeOf(cbrt)]` and `[BackwardDerivativeOf(cbrt)]`. You can refer to other existing functions.

Reply:
I see that the definition is just `pow(x, 0.33333)` now, in that case the perf difference should be small. It is still nicer to have custom derivatives defined, but the PR is good to merge.